lib/s3routes/routes/routeDELETE.ts

```diff
@@ -60,7 +60,7 @@ export default function routeDELETE(
                  */
                 if (err && (
                     !(err instanceof ArsenalError) || 
-                    (!err.is.NoSuchKey && err.is.NoSuchVersion)
+                    (!err.is.NoSuchKey && !err.is.NoSuchVersion)
```

Comment:
Can we add unit tests for these functions?

Reply:
I added some but they don't cover all the paths, I will add it (while I am confirming in cloudserver the fix)

Reply:
(we had 0 unit test for the routes before)

Reply:
Done in https://github.com/scality/Arsenal/pull/2273/commits/24a87f59fd4959bfebdc94bf0b32fb1d86bd0b27

we already have it but it was not testing both cases...